app/src/main/java/com/poupa/vinylmusicplayer/util/HouseKeeper.java

```diff
@@ -0,0 +1,111 @@
+package com.poupa.vinylmusicplayer.util;
+
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages a background thread to run async house-keeping tasks
+ *
+ * @author SC (soncaokim)
+ */
+public class HouseKeeper {
```

Comment:
can you use a more explicit name than that? especially since it's focused on threads

Reply:
You mean DelayedTaskThread?